plugins/modules/postgresql_pg_hba.py

```diff
@@ -364,7 +364,9 @@ def read(self):
                     line = line.strip()
                     comment = None
                     if '#' in line:
-                        line, comment = line.split('#', 1)
+                        result = re.search(r'((("([^"]|\")*"|([^#"])))*)(#(.*))?', line)
+                        comment = result.group(7)
+                        line = result.group(1)
```

Comment:
This regex does not catch all cases, for example
```
 a "b\"c#d" f # g # h " i
```

suggestion:
```python3
# on module load:
line_comment_regex = re.compile(r'(("(\\"|[^"])*"|[^"#])*)(#(.*))?')

# for every line:
parsed_line = line_comment_regex.fullmatch(line)
line = parsed_line.group(1)
comment = parsed_line.group(5)
```

Reply:
I was afraid, that I still miss a case :-( and there are even more code changes needed..
I'll check again later today